### DISCUSSION TITLE
show error in UI instead of silence

### DISCUSSION THREAD

**Discussion by dmotxilla** [danny-avila/LibreChat#4153] [Troubleshooting]
### What happened?

I developed a tool that returns too much information, then LibreChat tried to answer the prompt with the provided information from my tool, and there was a openai rate_limit_exceeded. However, the UI didn't show anything, just became silent, so the user does not understand what's going on.

### Steps to Reproduce

create a tool that returns too many information, and ask a prompt that uses the tool.

### Relevant log output

```shell
here it is a log of the error in the console:

`{"assistant_id":"asst_ofZyP7CTCWOhHRfMmYbQ86BB","cancelled_at":null,"completed_at":null,"created_at":1726792185,"expires_at":null,"failed_at":1726792194,"id":"run_EXncRjfHljb3CYTpBNID3SAW","incomplete_details":null,"instructions":"You are a virtual assistant for our company","last_error":{"code":"rate_limit_exceeded","message":"Request too large for gpt-4o in organization org-8cbmmCXeZMZeEHk2Bls8zcG0 on tokens per min (TPM): Limit 30000, Requested 43052. The input or output tokens must be reduced in order to run successfully.
`
```

### Screenshots

![image](https://github.com/user-attachments/assets/01f28049-57d0-4555-a45c-6990ac69516e)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Showing errors in the frontend is a bad security practice: https://owasp.org/API-Security/editions/2019/en/0xa7-security-misconfiguration/

While there could be better UX for your tool failing, the suggestion does not follow security best practices.

**Reply by dmotxilla** [reply]
i didn't mean to show the exact technical error as in the console. but show something, otherwise the user is waiting for the answer that never arrives.

**Reply by danny-avila** [reply]
In the case of assistants api, the error is logged in the backend, and the AI receives a message that the tool call failed. There should be better user experience, I agree, but some of it is in part due to how OpenAI handles this.
